Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -59,13 +58,13 @@ public class Channel {
   /// - Parameter host: an optional hostname override
   public init(address: String, certificates: String, host: String?) {
     self.host = address
-    underlyingChannel = cgrpc_channel_create_secure(address, certificates, host)
-    completionQueue = CompletionQueue(
-      underlyingCompletionQueue: cgrpc_channel_completion_queue(underlyingChannel), name: "Client")
+    underlyingChannel = cgrpc_channel_create_secure(address, certificates, &argumentValues, Int32(arguments.count))
+    completionQueue = CompletionQueue(underlyingCompletionQueue: cgrpc_channel_completion_queue(underlyingChannel), name: "Client")
     completionQueue.run() // start a loop that watches the channel's completion queue
   }
 
   deinit {
+    connectivityObservers.forEach { $0.polling = false }
```

Comment:
Either use a lock for setting `polling`, or use `$0.shutdown()` (see the other comments).